code/modules/unit_tests/designs.dm

```diff
@@ -0,0 +1,38 @@
+/datum/unit_test/designs
+
+/datum/unit_test/designs/Run()
+	var/datum/design/defaultDesign = new /datum/design()
+	var/datum/design/nanites/defaultDesignNanites = new /datum/design/nanites()
+	var/datum/design/surgery/defaultDesignSurgery = new /datum/design/surgery()
+
+	for(var/path in subtypesof(/datum/design))
+		if (ispath(path, /datum/design/nanites) || ispath(path, /datum/design/surgery)) //We are checking nanites and surgery design separatly later since they work differently
+			continue
+		var/datum/design/currentDesign = new path //Create an instance of each design
+		if (currentDesign.id == DESIGN_ID_IGNORE) //Don't check designs with ignore ID
+			continue
+		if (isnull(currentDesign.name) || currentDesign.name == defaultDesign.name) //Designs with ID must have non default/null Name
+			Fail("Design [currentDesign.type] has default or null name var but has an ID")
+		if ((!isnull(currentDesign.materials) && currentDesign.materials.len != 0) || (!isnull(currentDesign.reagents_list) && currentDesign.reagents_list.len != 0)) //Design requires materials
```

Comment:
Use `LAZYLEN`, it'll compress this down a lot.

Reply:
Changed